ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useMemo, useState } from 'react';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { TokenStandard } from '../../shared/constants/transaction';
+import { getCurrentChainId } from '../selectors';
+import {
+  getNFTContractInfo,
+  getTokenStandardAndDetails,
+} from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
+  value: string;
+};
+
+type CollectionsData = {
+  [key: string]: Collection;
+};
+
+// For now, we only support ERC721 tokens
+const SUPPORTED_NFT_TOKEN_STANDARDS = [TokenStandard.ERC721];
+
+export function useNftCollectionsMetadata(
+  requests: UseNftCollectionsMetadataRequest[],
+) {
+  const [collectionsMetadata, setCollectionsMetadata] =
+    useState<CollectionsData>({});
+  const chainId = useSelector(getCurrentChainId);
```

Comment:
Minor, we ideally need to move away from global network so this could be a property or at least a fallback.

Reply:
Done